specs/jsonschema-core.md

```diff
@@ -564,11 +581,11 @@ the keyword's value. Alternatively, an applicator may refer to a schema
 elsewhere in the same schema document, or in a different one. The mechanism for
 identifying such referenced schemas is defined by the keyword.
 
-Applicator keywords also define how subschema or referenced schema boolean
-[assertion](#assertions) results are modified and/or combined to produce the
-boolean result of the applicator. Applicators may apply any boolean logic
-operation to the assertion results of subschemas, but MUST NOT introduce new
-assertion conditions of their own.
+Applicator keywords also behave as assertions by defining how subschema or
+referenced schema boolean [assertion](#assertions) results are modified and/or
+combined to produce the boolean result of the applicator. Applicators may apply
+any boolean logic operation to the assertion results of subschemas, but MUST NOT
+introduce new assertion conditions of their own.
```

Comment:
> but MUST NOT introduce new assertion conditions of their own.

I'm not following this. What is this trying to say? What are we protecting against be forbidding it?

Reply:
That text was already there.

It's just saying that applicator assertions can only be combinatorial on their subschemas.  For instance `anyOf` only ORs the results of its subschemas without asserting anything else.

I think this is a simplicity guard, kind of a JSON Schema version of SRP.  I suppose this means something like `requiredProperties` would be in violation of the spec since it's combining subschema results _and_ asserting that all of the properties are there.

Reply:
I don't think it's unnecessary for that to be a constraint. Your `requiredProperties` example is great. Although I'm not a fan of that keyword, I don't think there should be any spec reason why it shouldn't be allowed.

> or maybe even an informative note suggesting simplicity over complexity?

Yes. I think that's a great idea.

Reply:
I've added a footnote and changes this to SHOULD NOT.